packages/enzyme-matchers/src/assertions/toHaveStyle.js

```diff
@@ -70,4 +70,25 @@ function toHaveStyle(
   };
 }
 
+function flattenStyle(style: ?any): ?Object {
+  if (!style) {
+    return undefined;
+  }
+
+  if (!Array.isArray(style)) {
+    return style;
+  }
+
+  var result = {};
+  for (var i = 0, styleLength = style.length; i < styleLength; ++i) {
+    var computedStyle = flattenStyle(style[i]);
+    if (computedStyle) {
+      for (var key in computedStyle) {
+        result[key] = computedStyle[key];
+      }
+    }
+  }
+  return result;
```

Comment:
This project gets full access to ES6 features. Could you re-write this with a `style.reduce` and use `const` or `let` where applicable? What I'm imagining:

```js
function flattenStyle(style) {
  //...
  return style.reduce((computedStyle, currentStyle) => {
    // logic
  }, {});
}

Reply:
Sure thing (my issue #190 made use of `reduce` and `filter`). I just thought of copying over the original implementation from `react-native` to get the exact same behaviour here. Anyways, the tests pass with the new implementation as well ;) Let me know what you think. I'm happy to further refine!